diff -uNr a/hpaftpd.c b/hpaftpd.c
--- a/hpaftpd.c	2007-07-01 09:47:45.000000000 +0000
+++ b/hpaftpd.c	2023-04-03 17:50:21.165917939 +0000
@@ -88,8 +88,8 @@
 #define CMD_BAD_OPTIONS		1
 #define CMD_VERSION		2
 
-//#define __DBG__		1		/* Enable debug mode (without deamon()) */
-//#define __DBG_USER__	1
+#define __DBG__		1		/* Enable debug mode (without deamon()) */
+#define __DBG_USER__	1
 
 #ifdef __DBG__
 #define __DBG(_arg, ...)	printf(_arg, ##__VA_ARGS__)
@@ -1405,6 +1405,7 @@
 //    }
 #endif
 
+    /*
     if(setregid(cfg.gid, cfg.gid)){
 	syslog(LOG_ERR, "Can't setgid user %s, %m", cfg.user);
 	return 6;
@@ -1413,6 +1414,7 @@
 	syslog(LOG_ERR, "Can't setuid user %s, %m", cfg.user);
 	return 7;
     }
+    */
 
     main_loop();
 
diff -uNr a/nb.h b/nb.h
--- a/nb.h	2007-06-18 21:02:03.000000000 +0000
+++ b/nb.h	2023-04-03 17:50:41.581918248 +0000
@@ -70,7 +70,7 @@
 extern NB_SERVER *nbs;
 extern NB_CONN *nbc;
 
-enum {
+typedef enum {
     NB_OK,		/* 0 - No errors */
     NB_ERROR,		/* 1 - Some error */
     NB_MEMORY,		/* 2 - Memory allocation error */
