diff --git a/boofuzz/utils/debugger_thread_simple.py b/boofuzz/utils/debugger_thread_simple.py
index cb41451..687e90c 100644
--- a/boofuzz/utils/debugger_thread_simple.py
+++ b/boofuzz/utils/debugger_thread_simple.py
@@ -227,6 +227,7 @@ class DebuggerThreadSimple(threading.Thread):
         Returns:
             bool: True if the target is still active, False otherwise.
         """
+        time.sleep(0.5)
         if self.is_alive():
             return True
         else:
